### DISCUSSION TITLE
Example with edge middleware and Sveltekit?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [vercel/community#3410] [Feature Question]
### Question

I would like to develop a "maintenance" page for my Sveltekit project deployed on Vercel.

Does anyone have documentation on how to "combine" Sveltekit and edge middleware?

**Comment by amyegan** [comment]
Hey @alexbjorlig. You can use [SvelteKit's server hooks](https://kit.svelte.dev/docs/hooks) to modify responses. Due to SvelteKit's client-side rendering, you cannot use Vercel's Edge Middleware with SvelteKit.

You could use the [redirect helper from `@sveltejs/kit`](https://kit.svelte.dev/docs/load#redirects) to specify a redirect location with a 3xx status code. Alternatively, you can set up [redirects](https://vercel.com/docs/concepts/projects/project-configuration#redirects) and [rewrites](https://vercel.com/docs/concepts/projects/project-configuration#rewrites) in `vercel.json`.

**Reply by alexbjorlig** [reply]
Hi @amyegan. Thanks for the response as always 👍

But, in Sveltekit the server hooks have this limitation:

<img width="656" alt="CleanShot 2023-07-28 at 17 29 23@2x" src="https://github.com/vercel/community/assets/9967422/ea5b31ea-e7d2-4d11-845f-839129c248d7">

So if I would like to implement something like a maintenance page, where all traffic is redirected based on for example edge config, it would not work for static assets.

That might be ok, as our team is already debating if it's "worth" static pre-rendering now that edge functions are so fast at rendering html.
